docs/content/tutorial/connecting-to-external-services.mdx

```diff
@@ -0,0 +1,184 @@
+---
+title: "Tutorial, part eight: Connecting to external services | Dagster Docs"
+description: Learn how to connect to external services in Dagster with resources
+---
+
+Throughout the tutorial, you built a data pipeline that connects to different external services, such as Hacker News. By building this asset graph, Dagster acts as a control plane of multiple different external services. However, connecting to these services and managing their use can be difficult. Resources are the recommended way to manage these connections in Dagster.
+
+Resources are Dagster objects that connect to external services, such as Slack, Snowflake, or Amazon S3. In this section, you’ll learn more about resources, including how to:
+
+- Define a resource and use it in an asset
+- Configure a resource
+- Customize a resource based on the environment where Dagster is running
+
+---
+
+## Why use resources?
+
+In the `topstory_ids` and `topstories` assets, you used the `requests` library to communicate with the Hacker News API. This code gets harder to maintain if:
+
+- Connections need to be shared across multiple assets
+- You need to authenticate with an external service securely
+- Configurations need to be consistent across its usage
+
+Aside from the complexity of configuring them, building a data pipeline creates many connections to external services. It becomes difficult to understand what assets use which connections or how often a connection is used.
+
+You should use resources to manage communicating with external services because they:
+
+- Enable flexibility in configuration
+- Standardize how connections are used across assets
+- Create a single source to monitor connections in the Dagster UI
+
+---
+
+## Step 1: Setting up a resource
+
+Suppose your Hacker News pipeline in Dagster has picked up momentum, and your stakeholders want to learn more about who is using it. You are tasked with analyzing how many people sign up for Hacker News.
+
+In the scaffolded Dagster project you made in Part 2 of the tutorial, you'll have seen a directory called `resources` with some files in it. The most important file is the `resources.py` file that exposes a resource called `DataGeneratorResource`. This resource generates simulated data about Hacker News signups. You'll use this resource to get the data needed to produce an asset for analysis.
```

Comment:
> Part 2 of the tutorial

🤔 Add a link here?

> you'll have seen a directory called `resources` with some files in it.

❓ Thoughts on rewording this a little? `you'll have seen` feels a little odd to me - what if I didn't? What about `you may have noticed`?